src/config.js

```diff
@@ -0,0 +1,1588 @@
+    if (options != null ? options.suppressException : undefined) {
+      try {
+        return this.makeValueConformToSchema(keyPath, value)
+      } catch (e) {
+        return undefined
+      }
+    } else {
+      let schema
+      if ((schema = this.getSchema(keyPath)) == null) {
+        if (schema === false) { throw new Error(`Illegal key path ${keyPath}`) }
+      }
+      return this.constructor.executeSchemaEnforcers(keyPath, value, schema)
+    }
+  }
+
+  // When the schema is changed / added, there may be values set in the config
+  // that do not conform to the schema. This will reset make them conform.
+  resetSettingsForSchemaChange (source) {
+    if (source == null) { source = this.getUserConfigPath() }
+    return this.transact(() => {
+      this.settings = this.makeValueConformToSchema(null, this.settings, {suppressException: true})
+      const selectorsAndSettings = this.scopedSettingsStore.propertiesForSource(source)
+      this.scopedSettingsStore.removePropertiesForSource(source)
+      for (let scopeSelector in selectorsAndSettings) {
+        let settings = selectorsAndSettings[scopeSelector]
+        settings = this.makeValueConformToSchema(null, settings, {suppressException: true})
+        this.setRawScopedValue(null, settings, source, scopeSelector)
+      }
+    })
+  }
+
+  /*
+  Section: Private Scoped Settings
+  */
+
+  priorityForSource (source) {
+    return (source === this.getUserConfigPath()) ? 1000 : 0
+  }
+
+  emitChangeEvent () {
+    if (this.transactDepth <= 0) { return this.emitter.emit('did-change') }
+  }
+
+  resetUserScopedSettings (newScopedSettings) {
+    const source = this.getUserConfigPath()
+    const priority = this.priorityForSource(source)
+    this.scopedSettingsStore.removePropertiesForSource(source)
+
+    for (let scopeSelector in newScopedSettings) {
+      let settings = newScopedSettings[scopeSelector]
+      settings = this.makeValueConformToSchema(null, settings, {suppressException: true})
+      const validatedSettings = {}
+      validatedSettings[scopeSelector] = withoutEmptyObjects(settings)
+      if (validatedSettings[scopeSelector] != null) { this.scopedSettingsStore.addProperties(source, validatedSettings, {priority}) }
+    }
+
+    return this.emitChangeEvent()
+  }
+
+  setRawScopedValue (keyPath, value, source, selector, options) {
+    if (keyPath != null) {
+      const newValue = {}
+      setValueAtKeyPath(newValue, keyPath, value)
+      value = newValue
+    }
+
+    const settingsBySelector = {}
+    settingsBySelector[selector] = value
+    this.scopedSettingsStore.addProperties(source, settingsBySelector, {priority: this.priorityForSource(source)})
+    return this.emitChangeEvent()
+  }
+
+  getRawScopedValue (scopeDescriptor, keyPath, options) {
+    scopeDescriptor = ScopeDescriptor.fromObject(scopeDescriptor)
+    const result = this.scopedSettingsStore.getPropertyValue(
+        scopeDescriptor.getScopeChain(),
```

Comment:
I think here and the one below this one were the indentation issues I was talking about.